src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -37,10 +41,12 @@
 (defn- build-rule
   "Build a new Rule based on the provided config-map"
   [config-map]
-  (let [rule-type (keyword (get config-map :type :path))]
-    (if (= rule-type :path)
-      (-> (rules/new-path-rule (config-map :path) (method config-map)))
-      (-> (rules/new-regex-rule (config-map :path) (method config-map))))))
+  (let [rule-type (keyword (get config-map :type :path))
+        rule-fn (rule-type new-rule-func-map)
+        rule (rule-fn (config-map :path) (method config-map))]
+    (if (true? (:allow_unauthenticated config-map))
+      (assoc rule :allow_unauthenticated true)
```

Comment:
> @jeffmccune Would that be an improvement?

I think so, I agree that having these transformation functions return a complete set of keys, even if the input configuration is lacking those keys, is an improvement.  It makes it much easier to reason about the output of these functions.